instrumentation/servlet/servlet-3.0/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/servlet/v3_0/Servlet3InitAdvice.java

```diff
@@ -20,7 +20,7 @@ public static void servletInit(
     if (servletConfig == null) {
       return;
     }
-    InstrumentationContext.get(Servlet.class, MappingResolver.class)
+    InstrumentationContext.get(Servlet.class, MappingResolverFactory.class)
         .putIfAbsent(servlet, new Servlet3MappingResolverFactory(servletConfig));
```

Comment:
😅 it took me a long time to understand how this PR changes anything in the init sequence, until I realized this line (which has no code changes) used to call `putIfAbsent(key, Supplier)` which immediately triggered `Servlet3MappingResolverFactory.getMappings()` via the super classes implementation of `Supplier.get()`, but now calls `putIfAbsent(key, non-supplier-value)` which does not